wfl/calculators/castep.py

```diff
@@ -95,17 +102,18 @@ def calculate(self, atoms=None, properties=_default_properties, system_changes=a
             # ASE castep calculator does not ever raise an exception when
             # it fails.  Instead, you get things like stress being None,
             # which lead to TypeError when save_calc_results calls get_stress().
-            for property in properties:
-                result = self.get_property(property)
+            for prop in properties:
+                result = self.get_property(prop, allow_calculation=False)
```

Comment:
Reason for the recursion - in `get_property` checks if results are present and if not, triggers a calculation. `allow_calculation` stops from running the calculation, but it's a dummy/unused argument in ASE v3.22, so there's no way of preventing the calculation via this argument.

Reply:
I take it that despite the CASTEP calculator being made more consistent with the rest of the ASE calculators, it still has the behavior that caused the need for this bit of code (i.e. `None` properties instead of raising an exception for failures)?

Reply:
I don't know which errors exactly weren't being raised.  Looking at the code, some warnings in 3.22 have been promoted to exceptions in 3.23, so we could assume that this issue has been fixed. I'd lean towards keeping this, though. It's a cheap check and should catch all the failures that still slipped through the update.

Reply:
sounds good